src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/mcoCe/McoCeExtractor.scala

```diff
@@ -0,0 +1,47 @@
+package fr.polytechnique.cmap.cnam.etl.extractors.mcoCe
+
+import java.sql.Timestamp
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.{DataFrame, Row}
+import fr.polytechnique.cmap.cnam.etl.events.{AnyEvent, Event, EventBuilder}
+import fr.polytechnique.cmap.cnam.etl.extractors.{EventRowExtractor, Extractor}
+import fr.polytechnique.cmap.cnam.etl.sources.Sources
+import fr.polytechnique.cmap.cnam.util.datetime.implicits._
+
+
+trait McoCeExtractor[EventType <: AnyEvent] extends Extractor[EventType] with McoCeSource with EventRowExtractor {
+
+  val columnName: String
+
+  val eventBuilder: EventBuilder
+
+  def getInput(sources: Sources): DataFrame = sources.mcoCe.get.select(ColNames.all.map(col): _*)
+
+  def isInStudy(codes: Set[String])
+    (row: Row): Boolean = codes.exists(code(row).startsWith(_))
+
+  def isInExtractorScope(row: Row): Boolean = !row.isNullAt(row.fieldIndex(columnName))
+
+  def builder(row: Row): Seq[Event[EventType]] = {
+    lazy val patientId = extractPatientId(row)
+    lazy val groupId = extractGroupId(row)
+    lazy val eventDate = extractStart(row)
+    lazy val endDate = extractEnd(row)
+    lazy val weight = extractWeight(row)
+
+    Seq(eventBuilder[EventType](patientId, groupId, code(row), weight, eventDate, endDate))
+  }
+
+  def code = (row: Row) => row.getAs[String](columnName)
+
+  def extractPatientId(r: Row): String = {
+    r.getAs[String](ColNames.PatientID)
+  }
+
+  override def extractGroupId(r: Row): String = {
```

Comment:
We need to discuss this subject. I imagine that we must agree on the role of a GroupId and stick to it for all the Extractors. This could be the object of an other PR in my opinion.

In this case, I would keep groupID = a stay identification (unique key within MCO_CE product of the PMSI).
What do you think ?

Reply:
I agree. `groupID` should always allow to __group__ Events that are related, for example through the same hospital stay. 

My documentation question is related to the fact that we should say what do you mean by your `groupID`, for instance here, the `groupID` stands for hospital stay ID.